src/overview/design-goals.md

```diff
@@ -19,6 +19,7 @@ We strive to start our protocol design process from user experience (UX) and use
 - The any hybrid PoW/PoS protocol (including PoW+TFL) block explorers will continue to function with the same UX through transitions in-so-far as displaying information about transactions, the mempool, and blocks.
 - Block explorers and other network metrics sites may require UX changes with respect to mining rewards and issuance calculations.
 - Network metrics sites may require UX changes with respect to the p2p protocol or other network-specific information.
+- Users can rely on [assured finality](../terminology.md#definition-assured-finality) with an expected time-to-finality of <30m.[^req-ttf-30m]
 
 ## Developer Experience Goals
```

Comment:
> - Wallet developers should not be required to make any changes through protocol transitions as long as they rely solely on the lightwalletd protocol or a full node API (such as the zcashd RPC interface).

I don't think we can guarantee this. The changes needed should be relatively minor, but some API changes could be needed. Certainly we *want* there to be some wallet changes to take advantage of new functionality, such as marking which transactions are finalized.

Reply:
> - For any hybrid PoW/PoS protocol (including PoW+TFL), mining pools and miners should not be required to make any software or protocol changes as long as they rely on zcashd-compatible GetBlockTemplate. One exception to this is software that bakes in assumptions about the block reward schedule, rather than relying on GetBlockTemplate solely.

I think this is also not necessarily realistic.

Reply:
> - For any hybrid PoW/PoS protocol (including PoW+TFL), the cost-of-attack to halt the chain should be larger than the 24 hour revenue of PoW mining rewards, given a “reasonably rigorous” security argument.

2/3rds of validator stake needs to be online for the BFT protocol not to halt. You'd need a *lot* of validators in order for the cost of a DDoS against a third of validator stake to be more than 24h of mining rewards (currently roughly USD 172800).

Reply:
Hm, when I wrote these I had two beliefs:

(a) They may be possible for some protocol designs / they are feasible goals.
(b) They may not be achievable with our specific design, as we introduce trade-offs.

This is why I added [this preface](https://github.com/Electric-Coin-Company/tfl-book/pull/105/files#diff-6803f4af6ae3ca1e7225a286076ea51449d9dcc4d27664593126396753d7820cR7-R9) in an earlier PR.

Do you support leaving that preface along with these idealized goals, even if we don't anticipate our design will achieve all of the ideal goals? The reason I think this is valuable is that elsewhere in-band we can add "trade-off asides".

For example, for light wallet development UX goals we may write something along the following lines (here "`$A`, `$B`, `$C`, and `$D`" are just example placeholder Goal references):

> Trade-off: This design approach achieves the security properties (goals `$A`, `$B`, and `$C`). However, because the transaction validity context necessary for $LOG_{da}$ differs from NU5 transaction validity context, wallet implementations that locally validate transactions against an NU5 (Bitcoin-style) context must be updated, so we explicitly trade off goal `$D` that aims to avoid altering wallet development APIs.

If we had that level of trade-off detail, we could then refactor the goals section to distinguish goals we believe we've achieved versus "aspirational goals we had to trade-off", and those could have back-links to the inline discussions about the specific trade-offs.

This would provide really deep understanding of our design approach, beyond just describing the design and properties: it would also describe how we got there and what choices were made along the way.

(I grant that the network DDOS resistance may be infeasible even as an idealized goal.)